src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -143,6 +121,55 @@ private void handleSiteList(RoutingContext rc) {
         }
     }
 
+    private static JsonObject createSiteObject(Site site, Map<Integer, List<LegacyClientKey>> clientKeys, List<LegacyClientKey> emptySiteKeys) {
+        JsonObject jo = new JsonObject();
+
+        JsonArray domainNamesJa = new JsonArray();
+        site.getDomainNames().forEach(domainNamesJa::add);
+
+        jo.put("id", site.getId());
+        jo.put("name", site.getName());
+        jo.put("description", site.getDescription());
+        jo.put("enabled", site.isEnabled());
+        jo.put("clientTypes", site.getClientTypes());
+        jo.put("domain_names", domainNamesJa);
+        jo.put("visible", site.isVisible());
+        jo.put("created", site.getCreated());
+
+        JsonArray jr = new JsonArray();
+        List<LegacyClientKey> clients = clientKeys.getOrDefault(site.getId(), emptySiteKeys);
+        clients.stream()
+                .map(LegacyClientKey::getRoles)
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet())
+                .forEach(jr::add);
+
+        jo.put("roles", jr);
+        jo.put("client_count", clients.size());
+        return jo;
+    }
+
+    private void handleSite(RoutingContext rc) {
```

Comment:
Maybe something like `handleSiteById` or `handleSiteGetById` would be a better name here? I know the existing names in this file aren't ideal. A newer endpoint in `GetClientSideKeypairsBySite.java` we use `handleGetClientSideKeys` but that is a slightly different pattern to this file. What do you think?

Reply:
Yeah that sounds good. I was just running with the pattern of the handleSiteList and went with that but that sounds better.